Libraries/MLXVLM/Models/Idefics3.swift

```diff
@@ -1112,17 +1129,20 @@ public class SmolVLMProcessor: UserInputProcessor {
             
             let videoPromptString = getVideoPromptString(frameCount: videoFrameResult.frames.count, timeStamps: videoFrameResult.timestamps, videoDuration: videoFrameResult.totalDuration, seqLen: imageSequenceLength, fakeToken: fakeImageToken, imageToken: imageToken, globalImageToken: globalImageToken)
             print(videoPromptString)
-            
-            let splitPrompt = decoded.split(by: imageToken)
-            let prompt = splitPrompt.joined(separator: videoPromptString)
+
+            // We need to insert after the system message, potentially
+            // With system message we have "<|im_start|>system_message<end_of_utterance>\nUser: ", without we have "<|im_start|>User: "
+            // For now we'll detect 'User'. We can improve with a regexp later
+            let splitPrompt = decoded.split(by: "User: ", options: .literal)
+            let prompt = splitPrompt[0] + "User: " + videoPromptString + splitPrompt[1]
```

Comment:
This is just what we needed to insert the list of frames in the prompt. I don't like that we use `User: `, we can improve later.